test/expressions/new-expr/new-on-domain.good

```diff
@@ -0,0 +1,2 @@
+new-on-domain.chpl:1: In function 'main':
+new-on-domain.chpl:2: error: invalid use of 'new' on chpl__buildDomainRuntimeType
```

Comment:
Ah okay.  I'd be tempted to open an issue on it with the label "easy" and see if anyone running through the compiler feels so moved.